contracts/BnomialToken1.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.2;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Burnable.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/draft-ERC20Permit.sol";
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Votes.sol";
+
+contract BinomialToken1 is ERC20, ERC20Burnable, Pausable, Ownable, ERC20Permit, ERC20Votes {
+    
+    
+    constructor() ERC20("Binomial Test Token", "BNT1") ERC20Permit("BinomialToken1") {
+        uint256 total_decimals = 100;
+        _mint(msg.sender, uint256(100 * 10 * total_decimals) );

```

Comment:
Why do you mint tokens in the constructor? I think we can start without any tokens and mint some to the treasury when needed.